### PR TITLE
325 react native wrapper

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport-internal#325

Ticket: 325

### Description

Create a react native wrapper around our Ios widget

### Release Plan

Merge once approved

### CODE REVIEW

File: packages/packages/react-native-sdk/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Metriport-iOS-SDK
```
Comment (Goncharo) [inline packages/packages/react-native-sdk/README.md:1]: `Metriport React Native SDK`

File: docs/more-info/react-native.mdx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        token="CONNECT_SESSION_TOKEN"
+        clientApiKey="CLIENT_API_KEY"
+        sandbox={false}
+        style={styles.box}
+      />
+    );
+  }
+
+  return <WebView source={{ uri: "https://connect.metriport.com/?token=demo" }} />;
+}
```
Comment (Goncharo) [inline docs/more-info/react-native.mdx:53]: General comment:

While we're at it, should we obfuscate this implementation from the end user?

ie `MetriportWidget` can be platform agnostic, and will handle all the URL params + logic internally.

So devs won't have to use raw URLs for Android, and will have a single interface/component for both platforms.

File: packages/packages/react-native-sdk/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+A React Native wrapper around our [iOS package](https://github.com/metriport/metriport-ios-sdk).
```
Comment (Goncharo) [inline packages/packages/react-native-sdk/README.md:3]: See previous comment, this could be better if it's just platform agnostic.

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: docs/more-info/react-native.mdx [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---
+title: "React Native"
+description: "A library to integrate with Metriport on React Native - including Apple Health integrations."
+---
+
+<Info>
+  As a prerequisite [visit the apple health docs](/more-info/apple-health#initial-setup) and take a
+  look at the initial setup for apple healthkit.
```
Comment (Goncharo) [inline docs/more-info/react-native.mdx:8]: "Before getting started, visit the [Apple Health docs](/more-info/apple-health#initial-setup) and follow the Initial Setup section to configure HealthKit."

File: docs/more-info/react-native.mdx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  in plain text to your repository.
+</Info>
+
+<Info>
+  As per the Quickstart guide, you'll need to generate a token on your server before opening the
+  Connect Widget. You can read more about this
+  [here](/getting-started/connect-quickstart#4-link-to-the-metriport-connect-widget-in-your-app).
+</Info>
+
+## Apple Healthkit
```
Comment (Goncharo) [inline docs/more-info/react-native.mdx:70]: Apple Health

File: docs/more-info/react-native.mdx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+To understand more how to receive data from apple healthkit [check out here.](/more-info/apple-health#receiving-data)
```
Comment (Goncharo) [inline docs/more-info/react-native.mdx:72]: "For details about receiving Apple Health data via webhooks, see the guide [here](/more-info/apple-health#receiving-data)."

File: packages/packages/react-native-sdk/metriport-react-native.podspec [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+Pod::Spec.new do |s|
+  s.name         = "metriport-react-native"
+  s.version      = package["version"]
+  s.summary      = package["description"]
+  s.homepage     = package["homepage"]
+  s.license      = package["license"]
+  s.authors      = package["author"]
+
+  s.platforms    = { :ios => "13.0" }
+  s.source       = { :git => "https://github.com/metriport.git", :tag => "#{s.version}" }
```
Comment (Goncharo) [inline packages/packages/react-native-sdk/metriport-react-native.podspec:15]: `https://github.com/metriport/metriport.git`  ?

File: packages/packages/react-native-sdk/src/index.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+      };
+
+export const MetriportWidget = (props: MetriportWidgetProps & WebViewProps) => {
+  if (Platform.OS === "ios") {
+    return <NativeWidget {...props} />;
+  }
+
+  const url = props.url ? props.url : "https://connect.metriport.com";
+
+  return <WebView source={{ uri: `${url}?token=${props.token}` }} {...props} />;
```
Comment (Goncharo) [inline packages/packages/react-native-sdk/src/index.tsx:38]: will this handle all the other params? ie:

- sandbox
- colorMode
- customColor
- providers

**Review by Goncharo — CHANGES_REQUESTED** [review]

File: packages/packages/react-native-sdk/metriport-react-native.podspec (hunk 7 above)
Comment (Orta21) [reply packages/packages/react-native-sdk/metriport-react-native.podspec:15]: I forgot an extra metriport but this is the url https://github.com/metriport/metriport.git it just reroutes to the repo but that seems to be the standard

https://guides.cocoapods.org/syntax/podspec.html#source

File: packages/packages/react-native-sdk/src/index.tsx [hunk 9]
```diff
@@ -33,7 +33,22 @@ export const MetriportWidget = (props: MetriportWidgetProps & WebViewProps) => {
     return <NativeWidget {...props} />;
   }
 
-  const url = props.url ? props.url : "https://connect.metriport.com";
+  let url = props.url ? props.url : "https://connect.metriport.com";
+
+  url = props.sandbox ? `${url}&sandbox=true` : url;
```
Comment (Goncharo) [inline packages/packages/react-native-sdk/src/index.tsx:38]: I think we'll need to change this so the first param is `https://connect.metriport.com?param=blah`, and every subsequent one has the `&`

**Review by Goncharo — CHANGES_REQUESTED** [review]

**Review by Goncharo — APPROVED** [review]
🚢 🚢 🚢 🚢

**Pull request merged**